src/components/pages/PendingApplications/ApplicationModal.js

```diff
@@ -1,27 +1,112 @@
 import React, { useEffect, useState } from 'react';
 import axiosWithAuth from '../../../utils/axiosWithAuth';
-import { Modal } from 'antd';
+import { Modal, Button, Popconfirm } from 'antd';
 import '../../../styles/styles.css';
+import './PendingApplication.css';
 
-const ApplicationModal = props => {
-  const [currentApplication, setCurrentApplication] = useState([]);
-  const { profileId, setProfileId, displayModal, setDisplayModal } = props;
+const ApplicationModal = ({
+  profileId,
+  setProfileId,
+  setDisplayModal,
+  displayModal,
+}) => {
+  const notes = { application_notes: '' };
+
+  const [currentApplication, setCurrentApplication] = useState({});
+  const [notesValue, setNotesValue] = useState(notes);
+  const [hideForm, setHideForm] = useState(true);
+
+  const updateModal = () => {
+    axiosWithAuth()
+      .get(`/application/profileId/${profileId}`)
+      .then(res => {
+        setCurrentApplication(res.data);
+      })
+      .catch(err => {
+        console.log(err);
+      });
+  };
 
   const handleOk = () => {
     setDisplayModal(false);
+    setDisplayModal(true);
   };
 
   const handleCancel = () => {
     setDisplayModal(false);
     setProfileId('');
+    setNotesValue(notes);
+    setHideForm(true);
+  };
+
+  const displayForm = () => {
+    setHideForm(false);
+  };
+  const handleChange = e => {
+    setNotesValue({
+      ...notesValue,
+      [e.target.name]: e.target.value,
+    });
+  };
+  const addNote = e => {
+    axiosWithAuth()
+      .put(
+        `/application/update-notes/${currentApplication.application_id}`,
+        notesValue
+      )
+      .then(res => {
+        updateModal();
+      })
+      .catch(err => {
+        console.log(err);
+      });
+    e.preventDefault();
+    setHideForm(true);
+  };
+
+  /**
+   * Author: Khaleel Musleh
+   * @param {approveApplication} e is for approving an application of a mentor_intake or mentee_intake Boolean from false to approved:true making a PUT call to the backend database server.
+   */
+
+  const approveApplication = e => {
```

Comment:
Nice job getting the event handlers set up to approve and reject applications. Code looks clean.

Reply:
Thank you Kim 😊